### DISCUSSION TITLE
Spells being Proc'd while wearing equipment.

### DISCUSSION THREAD

**Discussion by HMxMisfit** [AscensionGameDev/Intersect-Engine#2094] [Ideas]
I would like to be able to cast spells with a percentage chance while wearing equipment. 

So for example. If you were to use a dagger you could have a 10% chance to cast double strike while attacking. Or if you were to wear armor you could create a 10% chance to cast a buff on yourself while being attacked. 

The Proc would work when health damage was received or given.

**Comment by PyroTech03** [comment]
I can see this greatly helping differentiate equipment.

**Comment by Mighty-Professional** [comment]
Seems like a worthwhile feature. Might not make sense for some spells but that is up to the game designer I suppose. Might require work if spells require targets.

**Reply by PyroTech03** [reply]
It'd allow you to say gave a sword that has a chance to cause bleed or poison on target. Onus would be on the designer to decide what can or cannot apply.

#2181 could almost get there for this and cover armor. But I don't think it'd cover weapons. :L

**Reply by HMxMisfit** [reply]
Yeah. I was thinking at the time it would just target the entity that is being attacked automatically. The main reason I wanted this feature was to create more unique equipment besides as a means to boosts stats. This would make grinding bosses or crafting more rewarding overall and have more potential for character build variety.

**Reply by PyroTech03** [reply]
If the onhit property in #2181 can translate that'd cover weapons as well. Then you wouldn't need an additional field/setting.

Or just allow you to select a spell in an onhit thing. I think that's what the extra effect option is supposed to allow but I've not seen anyone describe proper use/setup for it.

**Reply by HMxMisfit** [reply]
While that's true. But, having spell procs on equipment would be part of the description. So when reading the equipment for example you'll see: 
+1 strength
 ... 
 10% chance to cast double strike on-hit. 

Attaching events to the items wouldn't allow this. Although they share similarities.

**Reply by PyroTech03** [reply]
I see what you mean.
